staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -316,6 +317,10 @@ func (s *SecureServingOptions) MaybeDefaultWithSelfSignedCerts(publicAddress str
 		if cert, key, err := certutil.GenerateSelfSignedCertKeyWithFixtures(publicAddress, alternateIPs, alternateDNS, s.ServerCert.FixtureDirectory); err != nil {
 			return fmt.Errorf("unable to generate self signed cert: %v", err)
 		} else if len(keyCert.CertFile) > 0 && len(keyCert.KeyFile) > 0 {
+			// Ensure that the key matches the cert and both are valid
+			if _, err := tls.X509KeyPair(cert, key); err != nil {
```

Comment:
These have just been generated by our own code (four lines up).  It doesn't seem likely that we generated invalid certificates.  Is this being created because of a bug you observed?